svm/src/account_loader.rs

```diff
@@ -1391,6 +1393,148 @@ mod tests {
         assert_eq!(result.err(), Some(TransactionError::AccountNotFound));
     }
 
+    #[test]
+    fn test_load_transaction_accounts_program_account_executable_bypass() {
```

Comment:
thats correct. the issue is the current code just checks if something exists in the cache, and then assumes its executable, so it attempts to execute and it fails. it does load the data anyway (so using the cache isnt really an optimization) but just to see if it exists, which i assume was added to fix a bug

```rust
    } else if let Some(program) = (!is_instruction_account && !is_writable)
        .then_some(())
        .and_then(|_| loaded_programs.find(account_key))
    {
        callbacks
            .get_account_shared_data(account_key)
            .ok_or(TransactionError::AccountNotFound)?;
        // Optimization to skip loading of accounts which are only used as
        // programs in top-level instructions and not passed as instruction accounts.
        LoadedTransactionAccount {
            loaded_size: program.account_size,
            account: account_shared_data_from_program(&program),
            rent_collected: 0,
        }
    } else {
```

checking `executable` on the account here requires a feature gate because (until fee-only txn are enabled) it would change whether the transaction pays fees and thus affects consensus. but i talked with pankaj the other week and he said that the status on the cache item is sufficient to determine if it is in fact executable or not (this is what i meant in the tombstone comment, ill expand it in next commit). so in the future when we fix this i believe we can just do this

```rust
        } else if let Some(ref program) = (!is_instruction_account && !is_writable)
            .then_some(())
            .and_then(|_| program_cache.find(account_key))
            .filter(|program| !program.is_tombstone())
        {   
            loaded_accounts_map.insert_cached_program(*account_key, program);
        } else if ... {
```